src/core/Directus/Services/TablesService.php

```diff
@@ -735,6 +743,29 @@ public function getTableObject($tableName)
         return SchemaService::getCollection($tableName);
     }
 
+    public function hasPrimaryField($fields){
+        $result = false;
+
+        foreach($fields as $field){
+            if($field['primary_key'] === true){
+                $result = true;
+                break;
+            }
+        }
+        return $result;
+    }
+
+    public function hasUniquePrimaryKey($fields){
```

Comment:
Adding the type of `$fields` in the function declaration parameters make it avoid us to add validation to the function.

Example: `public function hasUniquePrimaryKey(array $fields)`